static/app/utils/profiling/spanTree.tsx

```diff
@@ -44,21 +37,24 @@ class SpanTreeNode {
 }
 
 class SpanTree {
-  spans: RawSpanType[];
   spanTree: SpanTreeNode = SpanTreeNode.Root();
   orphanedSpans: RawSpanType[] = [];
 
   constructor(spans: RawSpanType[]) {
-    this.spans = spans;
-    this.buildCollapsedSpanTree();
+    this.buildCollapsedSpanTree(spans);
+
+    this.spanTree.span.timestamp = this.spanTree.children.reduce(
+      (sum, c) => sum + c.span.timestamp - c.span.start_timestamp,
+      0
+    );
   }
 
   static Empty(): SpanTree {
     return new SpanTree([]);
   }
 
-  buildCollapsedSpanTree() {
-    const spansSortedByStartTime = [...this.spans].sort((a, b) => {
+  buildCollapsedSpanTree(spans: RawSpanType[]) {
+    const spansSortedByStartTime = [...spans].sort((a, b) => {
```

Comment:
I know this is merged already but if I'm reading this correctly, is this constructing the span tree by checking if one span is contained within another based on the timestamps? If so, this is not always correct since a span being contained within another does not guarantee it's a child of the other. What dictates the parent-child relationship between spans is the `span_id` and `parent_span_id` on each span.

Reply:
Yes, that's the idea for now - try and fill as much space as we can with the spans. Once things start rendering, I think we will update the orphaned check to actually check for span_id/parent_span_id edge and orphan spans based on that.

Reply:
Okay, let's leave a comment indicating this is the case so it's clear that this current iteration intentionally doesn't use the span ids.